src/cluster_legacy.c

```diff
@@ -1470,15 +1470,28 @@ int clusterNodeRemoveSlave(clusterNode *master, clusterNode *slave) {
 }
 
 int clusterNodeAddSlave(clusterNode *master, clusterNode *slave) {
-    int j;
+    int j, pos = -1;
 
     /* If it's already a slave, don't add it again. */
-    for (j = 0; j < master->numslaves; j++)
+    for (j = 0; j < master->numslaves; j++) {
         if (master->slaves[j] == slave) return C_ERR;
-    master->slaves = zrealloc(master->slaves,
-        sizeof(clusterNode*)*(master->numslaves+1));
-    master->slaves[master->numslaves] = slave;
+        /* Find lexicographic order index for the slave. */
+        if (pos == -1 && strcasecmp(slave->name, master->slaves[j]->name) > 0) {
```

Comment:
Ideally 
```suggestion
        if (pos == -1 && strncasecmp(slave->name, master->slaves[j]->name, CLUSTER_NAMELEN) > 0) {
```

Since these strings are not case sensitive.